### DISCUSSION TITLE
Stop Loss Placement Issue in Strategy

### DISCUSSION THREAD

**Discussion by crueltytanya** [kernc/backtesting.py#1123] [Q&A]
Hello, I'm experiencing difficulty with setting up SL (Stop Loss). I set the price for SL as follows: for LONG positions, it's the low price of the previous candle, and for SHORT positions, it's the high price of the previous candle. However, in the majority of cases, the SL works correctly, but for 35% of the trades with SL, the value is incorrect, and it's skewing the performance metrics of the strategy. Could you please advise on what might be the issue?
![Снимок](https://github.com/kernc/backtesting.py/assets/135027177/de22ed0c-2194-4632-8590-807a00bbff5f)

my code:
`
self.buy(sl=self.data.Low[-1])
self.sell(sl=self.data.High[-1])
`

**Comment by s-kust** [comment]
Check if you have `trade_on_close` True or False. As far as I understand, if this parameter is False, then the stop loss order will be executed the next day at the opening price, the same as all other orders.

You may also find this - [https://github.com/s-kust/python-backtesting-template/blob/main/utils/strategy_exec/stop_loss.py](https://github.com/s-kust/python-backtesting-template/blob/main/utils/strategy_exec/stop_loss.py) - helpful
